modules/pam_faillock/pam_faillock.c

```diff
@@ -478,6 +488,10 @@ pam_sm_authenticate(pam_handle_t *pamh, int flags,
 		goto err;
 	}
 
+	rv = pam_set_data(pamh, FAILLOCK_AUTH_EXECUTED, (void *)1, set_data_cleanup);
```

Comment:
If the cleanup function is no-op, a null pointer could be specified instead of the cleanup function.